### PR TITLE
fix: improved design and logic for M1

### PR DESCRIPTION
**Pull request by Fyuryy**
# Fixing design and adding some logic 

What was done?
- Replaced Interest grid by FlatList component for performance reasons
- Fixed textbox bug that didn't allow the user to click on the boxes
- Added feature allowing the user to quit the keyboard by clicking on the screen
- Added feature allowing the user to go to the next input box by clicking 'Enter'
- Corrected selection colors in the interest selection list
- Fonts load before any screen is displayed

What is there to do during the next milestone?
- Add date picking feature to allow the user to input their birth date in an easier and more intuitive way.
- Interactions with Fire base so that users can create accounts properly 
- Code the rest of the screens for the 
<p float="left">

# Unselected and selected interests

</p>

### CODE REVIEW

**Review by GaelCondeLosada — CHANGES_REQUESTED** [review]
I didn't have the time to check all the code in details but there are a few things that would require changes in my opinion.

- First of all I suggest you can delete the file `AppContent.tsx` as it is not a part of the app.

- Secondly on Android there are a few problems:
After clicking on Add a description now, the button still says "Add a description now" and the bottom navigation bar is slightly cut out. However I think the navigation bar being slightly cut out is not a top priority problem to fix.


Another problem that's a bit more concerning is when trying to type a description, the keyboard goes over the description section.

File: screens/Onboarding/OnboardingScreen.tsx [hunk 1]
```diff
@@ -13,66 +21,76 @@ const OnboardingScreen: React.FC = () => {
...
-        placeholder="Username or email"
-        placeholderTextColor={'black'}
-        keyboardType="email-address"
-        autoCapitalize="none"
-      />
+        {/* Username/Email Field */}
+        <TextInput
+          style={[styles.inputField, globalStyles.text]}
+          placeholder="Username or email"
+          placeholderTextColor={'black'}
```
Comment (GaelCondeLosada) [inline screens/Onboarding/OnboardingScreen.tsx:37]: I think you should use the colors.ts file in assets for the colors.

File: screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx [hunk 2]
```diff
@@ -49,83 +56,85 @@ const AuthenticationScreen: React.FC = () => {
...
+        <InputField
+          label="Confirm Password*"
+          placeholder="****************"
+          onChangeText={setConfirmPassword}
+          value={confirmPassword}
+        ></InputField>
+
+        <View style={styles.container}>
+          <View style={styles.phrase}>
+            <Entypo name="cross" color="red" />
```
Comment (GaelCondeLosada) [inline screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx:84]: Same thing for the colors. There are other ones too.

File: App.tsx [hunk 3]
```diff
@@ -18,28 +18,43 @@ export default function App() {
...
     async function prepare() {
-      await SplashScreen.preventAutoHideAsync();
+      try {
+        await SplashScreen.preventAutoHideAsync();
+        await new Promise((resolve) => setTimeout(resolve, 2000));
+      } catch (e) {
+        console.warn(e);
+      } finally {
+        setAppIsReady(true);
+      }
```
Comment (AlbertoCentonze) [inline App.tsx:32]: Can this fail? Do we need a try/catch block here?

File: components/DatePicker/DatePicker.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  return (
+    <Pressable style = {styles.absView} onPress={() => setDateModal(false)}>
+      <View style={styles.dateContainer}>
+        <DateTimePicker
+          // minimumDate={}
+          // maximumDate={}
+          testID="dateTimePicker"
+          value={date}
+          mode="date"
+          display={Platform.OS === 'ios' ? 'spinner' : 'default'}
```
Comment (AlbertoCentonze) [inline components/DatePicker/DatePicker.tsx:37]: Was this properly tested on android?

File: components/InputField/InputField.tsx [hunk 5]
```diff
@@ -1,33 +1,36 @@
...
   return (
     <View style={styles.section}>
-      <Text style={[styles.topText, globalStyles.text]}>{label}</Text>
+      <Text style={[styles.label, globalStyles.text]}>{label}</Text>
       <TextInput
         placeholder={placeholder}
         style={[styles.input, globalStyles.text]}
         value={value}
         onChangeText={onChangeText}
+        onSubmitEditing={onSubmitEditing}
```
Comment (AlbertoCentonze) [inline components/InputField/InputField.tsx:29]: Do we need `forwaredRef` for this to work?

File: screens/Registration/InterestsScreen/InterestsScreen.tsx [hunk 6]
```diff
@@ -41,7 +44,7 @@ const InterestButton: React.FC<InterestButtonProps> = ({
 
 const interests = [
   'Machine Learning',
-  'Artificial Intelligence',
+  'Artificial Intelligence is very cool',
```
Comment (AlbertoCentonze) [inline screens/Registration/InterestsScreen/InterestsScreen.tsx:47]: I think this was a test but you forgot to undo it ;)

File: screens/Registration/InterestsScreen/InterestsScreen.tsx [hunk 7]
```diff
@@ -53,6 +56,11 @@ const interests = [
   'Ethereum',
   'Solana',
   'Computer graphics',
+  'Bananas',
+  'Apples',
+  'Big tech',
+  'Finance',
+  'Gaspard tes teubé t con',
```
Comment (AlbertoCentonze) [inline screens/Registration/InterestsScreen/InterestsScreen.tsx:63]: Please remove this

**Review by AlbertoCentonze — CHANGES_REQUESTED** [review]
Overall a very nice PR, let's try to make more smaller PRs next time as we discussed during the retrospective! Also I would recommend already removing all the semi-columns everywhere since this will come up as a conflict during merge.

File: App.tsx (hunk 3 above)
Comment (Fyuryy) [reply App.tsx:32]: I think it's better if we let it for safety reasons

File: components/DatePicker/DatePicker.tsx (hunk 4 above)
Comment (Fyuryy) [reply components/DatePicker/DatePicker.tsx:37]: yes!

File: components/InputField/InputField.tsx (hunk 5 above)
Comment (Fyuryy) [reply components/InputField/InputField.tsx:29]: forwardRef is necessary when you want to pass a ref down to a child component that is not directly accessible due to being encapsulated within another component. It allows the child component to expose its own ref to the parent component. Without forwardRef, the ref you pass to the child would reference the instance of the child component itself, not the underlying DOM element or native control you're trying to access.

File: screens/Registration/InterestsScreen/InterestsScreen.tsx (hunk 7 above)
Comment (Fyuryy) [reply screens/Registration/InterestsScreen/InterestsScreen.tsx]: Done

**Review by AlbertoCentonze — APPROVED** [review]
LGTM!

**Review by GaelCondeLosada — APPROVED** [review]
Looks pretty good to me now, the layout is still not perfect with the keyboard on android but the functionality work fine. Date picker is functional on android. It is good enough for me we ca make another PR to fix small details later.

**Pull request merged**